Services/Statistics/SimulationStatistics.cs

```diff
@@ -99,6 +102,40 @@ public async Task CreateOrUpdateAsync(string simulationId, SimulationStatisticsM
             }
         }
 
+        public async Task DeleteSimulationStatisticsAsync(string simulationId)
+        {
+            if (string.IsNullOrEmpty(simulationId))
+            {
+                this.log.Error("Simulation Id cannot be null or empty");
```

Comment:
I think this should throw an exception, or is it expected that the Id can be null?

Reply:
Id cannot be null, added exception